src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,117 +1,227 @@
+    // `aria-label` for screen readers.
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      !label &&
+      !props['aria-label']
+    ) {
+      throw new Error('You must provide a visible label or aria-label');
+    }
+    const generatedId = useUID();
+    const checkboxId = id || generatedId;
+
+    return (
+      <div className={clsx(className, styles['checkbox'])}>
+        <CheckboxInput id={checkboxId} ref={ref} {...other} />
+        {label && <Label htmlFor={checkboxId} size={size} text={label} />}
+      </div>
+    );
+  },
+);
 
+Checkbox.displayName = 'Checkbox';
+
+/**
+ * Checkbox input element, exported for greater flexibility.
+ * You must provide an `id` prop and connect it to a visible label.
+ */
+const CheckboxInput = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ checked, className, disabled, ...other }, ref) => {
+    // Make indeterminate checkbox visually match the colors of a
+    // checked state, but announce itself as "mixed" to screen readers
+    const checkedProps =
+      checked === 'indeterminate'
+        ? {
+            'aria-checked': 'mixed' as const,
+            checked: true,
+          }
+        : {
+            checked,
+          };
+
+    return (
+      <span
+        className={clsx(
+          styles['input__wrapper'],
+          disabled && styles['input__wrapper--disabled'],
+        )}
+      >
+        <input
+          className={clsx(className, styles['checkbox__input'])}
+          data-bootstrap-override="checkbox"
+          disabled={disabled}
+          ref={ref}
+          style={svgStyle}
+          type="checkbox"
+          {...checkedProps}
+          {...other}
+        />
+        <CheckboxSvg indeterminate={checked === 'indeterminate'} />
+      </span>
+    );
+  },
+);
+
+CheckboxInput.displayName = 'CheckboxInput';
+
+/**
+ * Label element, exported for greater flexibility. Can be used with any form input.
+ */
+const Label = ({ text, className, htmlFor, size = 'medium' }: LabelProps) => {
   return (
-    <div className={componentClassName} {...other}>
-      <input
-        id={id}
-        ref={checkboxRef}
-        type="checkbox"
-        name={name}
-        value={value}
-        checked={checked}
-        className={styles['checkbox__input']}
-        disabled={disabled}
-        readOnly={readOnly}
-        onChange={onChange && onChange}
-        aria-describedby={ariaDescribedBy}
-        onKeyDown={(e) => onKeyDown(e)}
-      />
-      <span className={styles['checkbox__custom-check']} />
-    </div>
+    <label
+      className={clsx(
+        className,
+        styles['label'],
+        size === 'small' && styles['label--sm'],
+        size === 'medium' && styles['label--md'],
+      )}
+      data-bootstrap-override="label"
+      htmlFor={htmlFor}
+      style={svgStyle}
+    >
+      {text}
+    </label>
   );
 };
+
+export default Checkbox;
+export { Checkbox, CheckboxInput, Label };
```

Comment:
I believe we decided to stop doing `export` default so we could replace the `const` declarations above with `export const`

Reply:
Sounds good, will remove and update exports